UpdateData/src/UpdateData.java

```diff
@@ -0,0 +1,205 @@
+import java.util.ArrayList;
+import java.util.List;
+import org.bson.Document;
+//import org.bson.conversions.Bson;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Filters.and;
+//import static com.mongodb.client.model.Filters.lt;
+import static com.mongodb.client.model.Filters.gt;
+
+public class UpdateData {
+
+	/**
+	 * Updates documents in the collection using updateOne()
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param total: Total of the document that is to be updated.
+	 * @param subtotal: The subtotal of the document whose total is passed is updated with this value.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	
+	public List<Document> UpdateOneOrder(String connectionString, Double total, Double subtotal)
+	{
+
+		if(connectionString == null || connectionString.isEmpty() || total <= 0 || subtotal <= 0)
+		{
+			throw new IllegalArgumentException();
+		}
+		
+		MongoClientURI clientUri = new MongoClientURI(connectionString);
+		try(MongoClient client = new MongoClient(clientUri))
+		{
+		
+			MongoDatabase database = client.getDatabase("stores");
+			MongoCollection<Document> collection = database.getCollection("orders");	
+	
+			Document newDocument = new Document();
+			newDocument.append("$set", new Document().append("subtotal", subtotal));
+
+			Document updateQuery = new Document().append("total", total);
+
+			collection.updateOne(updateQuery, newDocument);
+			
+	        List<Document> queryResult = collection
+	        		.find(and(eq("total",total),eq("subtotal",subtotal)))
+	        		.into(new ArrayList<Document>());
+			return queryResult;
+		}
+		
+		catch (Exception e) {
+			//log the exception
+			
+			System.out.println("An exception occured");
+			System.out.println("Details:");
+			System.out.println(e.getStackTrace());
+			return null;
+							
+		}
+	}
+	
+	/**
+	 * Updates documents in the collection using updateOne()
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param sku: lineitems - sku of the document that is to be updated.
+	 * @param name: The lineitems - name of the document whose sku is passed is updated with this value.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	
+	public List<Document> UpdateOneOrderForEmbeddedField(String connectionString, String stockKeepingUnit, String name)
```

Comment:
give better name than the function. It could be 'updateitemname' or something relevant